sqlglot/dialects/bigquery.py

```diff
@@ -385,6 +385,25 @@ def _parse_table_parts(self, schema: bool = False) -> exp.Table:
 
             return table
 
+        def _parse_json_object(self) -> exp.JSONObject:
+            expressions: t.List[exp.Expression] = []
+            while True:
+                key = self._parse_field()
+                if key is None:
+                    if not expressions:
+                        break
+                    self.raise_error("Expecting key argument")
+                if not self._match(TokenType.COMMA):
+                    self.raise_error("Expecting a matching value argument")
+                value = self._parse_expression()
+                if value is None:
+                    self.raise_error("Expecting value argument")
+                expressions.append(self.expression(exp.JSONKeyValue, this=key, expression=value))
+                if not self._match(TokenType.COMMA):
+                    break
+
+            return self.expression(exp.JSONObject, expressions=expressions)
+
```

Comment:
This whole method seems unnecessary, it probably suffices to do:

```diff
+++ b/sqlglot/parser.py
@@ -4139,9 +4139,9 @@ class Parser(metaclass=_Parser):
     def _parse_json_key_value(self) -> t.Optional[exp.JSONKeyValue]:
         self._match_text_seq("KEY")
         key = self._parse_field()
-        self._match(TokenType.COLON)
+        self._match_set((TokenType.COLON, TokenType.COMMA))
         self._match_text_seq("VALUE")
-        value = self._parse_field()
+        value = self._parse_column()
```

Reply:
What problem does this cause in this case?

Reply:
This parse all arguments with comma separated and apply the `self._parse_json_key_value` method, there won't be a key and a value inside the `_parse_json_key_value`, but only one of them.

Reply:
I'm not sure if this is a real issue, iirc the variants that `_parse_json_object` is supposed to cover are `JSON_OBJECT(k:v [, ...])` and `JSON_OBJECT(KEY k VALUE v [, ...])`. To this point, I also didn't get any test failures locally after applying the suggested changes.

Reply:
Ah yes, you are right. I thought _parse_csv would split and passes the items, but it just splits and calls the next parse method. Your patch then looks indeed much better. Let me use that.